kubernetes-kit-starter/src/main/java/com/vaadin/kubernetes/starter/sessiontracker/serialization/debug/Job.java

```diff
@@ -238,15 +243,17 @@ Result complete() {
         }
         long duration = TimeUnit.NANOSECONDS
                 .toMillis(System.nanoTime() - this.startTimeNanos);
-        messages.computeIfPresent(Outcome.NOT_SERIALIZABLE_CLASSES.name(),
-                (unused, info) -> info.stream()
-                        .flatMap(className -> Stream.concat(
-                                Stream.of(className),
-                                unserializableDetails
-                                        .getOrDefault(className,
-                                                Collections.emptyList())
-                                        .stream().map(entry -> "\t" + entry)))
-                        .collect(Collectors.toList()));
+        if (!outcome.contains(Outcome.SERIALIZATION_TIMEOUT)) {
+            messages.computeIfPresent(Outcome.NOT_SERIALIZABLE_CLASSES.name(),
+                    (unused, info) -> info.stream()
+                            .flatMap(className -> Stream.concat(
+                                    Stream.of(className),
+                                    unserializableDetails
+                                            .getOrDefault(className,
+                                                    Collections.emptyList())
+                                            .stream().map(entry -> "\t" + entry)))
+                            .collect(Collectors.toList()));
+        }
         return new Result(sessionId, storageKey, outcome, duration, messages);
```

Comment:
I wonder if there could be any other concurrent modification exception, if accessing messages map in `Result` while the serialization process is still running.
What about making a defensive copy of the map in the `Result` constructor?

Reply:
Yes, that makes sense.  Created a copy of messages in the `Result` constructor.